mig/shared/install.py

```diff
@@ -944,20 +944,40 @@ def generate_confs(
     user_dict['__DUPLICATI_PROTOCOLS__'] = ' '.join(prio_duplicati_protocols)
 
     if timezone == keyword_auto:
-        # attempt to detect the timezone
+        # Attempt to detect the timezone in various known portable ways
         sys_timezone = None
-        try:
-            timezone_cmd = ["/usr/bin/timedatectl", "status"]
-            timezone_proc = subprocess_popen(
-                timezone_cmd, stdout=subprocess_pipe)
-            for line in timezone_proc.stdout.readlines():
-                line = ensure_native_string(line.strip())
-                if not line.startswith("Time zone: "):
-                    continue
-                sys_timezone = line.replace("Time zone: ", "").split(" ", 1)[0]
-        except OSError as exc:
-            # warn about any issues executing the command but continue
-            pass
+
+        env_timezone = os.environ.get('TZ', None)
```

Comment:
I shared some thoughts about a number of items offline, a decision made because a lot of that commentary was speculative and could well be slightly further work territory.

But, one specific thing I do think wants to be addressed is placing this logic into its own function in the file - I'd suggest a name like `def _determine_timezone(): ...` which would a) service to contain it and b) make further iteration more straightforward.

Reply:
Yes, that sounds like a better way forward :+1: